test/getGitWorkspaceState.test.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as fse from 'fs-extra';
+import { join } from 'path';
+import { Uri } from "vscode";
+import { parseError } from 'vscode-azureextensionui';
+import { getGitWorkspaceState, GitWorkspaceState, promptForDefaultBranch, tryGetDefaultBranch, verifyGitWorkspaceForCreation } from "../extension.bundle";
+import { createTestActionContext, testFolderPath, testUserInput } from "./global.test";
+
+suite('Workspace Configurations for SWA Creation', function (this: Mocha.Suite): void {
+    this.timeout(2 * 60 * 1000);
+    const context = createTestActionContext();
+    const testFolderUri: Uri = Uri.file(testFolderPath);
+
+    test('Empty workspace with no git repository', async () => {
+        const gitWorkspaceState: GitWorkspaceState = await getGitWorkspaceState(context, testFolderUri)
+        let message: string = '';
+        try {
+            await verifyGitWorkspaceForCreation(context, gitWorkspaceState, testFolderUri)
+        } catch (err) {
+            const pError = parseError(err);
+            message = pError.message;
+        }
+
+        assert.strictEqual(message, 'Cannot create a Static Web App with an empty workspace.')
+    });
+
+    test('Workspace with no git repository', async () => {
+        await fse.writeFile(join(testFolderPath, 'test.txt'), 'Test');
+
+        const gitWorkspaceState: GitWorkspaceState = await getGitWorkspaceState(context, testFolderUri);
+        assert.strictEqual(gitWorkspaceState.repo, null, `Workspace contained a repository prior to test "${gitWorkspaceState.repo?.rootUri.fsPath}"`);
+        const testCommitMsg: string = 'Test commit';
+        await testUserInput.runWithInputs(['Create', testCommitMsg], async () => {
+            await verifyGitWorkspaceForCreation(context, gitWorkspaceState, testFolderUri);
+        });
+
+        assert.ok(gitWorkspaceState.repo);
```

Comment:
It's a bit weird to me that `verifyGitWorkspaceForCreation` modifies the original `gitWorkspaceState` object. I would've expected the test needed to call `getGitWorkspaceState` again